packages/api/src/models/medical/__tests__/patient.ts

```diff
@@ -2,5 +2,14 @@ import { Patient } from "@metriport/core/domain/patient";
 import { makePatient } from "../../../domain/medical/__tests__/patient";
 import { PatientModel } from "../patient";
 
-export const makePatientModel = (params?: Partial<Patient>): PatientModel =>
+export const makePatientAsPatientModel = (params?: Partial<Patient>): PatientModel =>
   makePatient(params) as PatientModel;
+
+export const makePatientModel = (params?: Partial<PatientModel>): PatientModel => {
+  const patient = makePatient(params);
+  const model = new PatientModel(patient);
+
+  model.data = patient.data;
+
+  return model;
+};
```

Comment:
Its not so much mocking the `getPatientOrFail` the reason i need to use the PatientModel is because we call `patient.update()`  i could change this `PatientModel.update()` but didnt want to change the logic to make the test easier.

As for the last question ill make the one i created the default one and remove `makePatientAsPatientModel` as its hacky and doesnt contain the other aspects of a model like in this case being able to call update on the returned value

Reply:
WRT to calling `patient.update()`, could just use a mock object like so:

```ts
const mockPatient = {
  update: jest.fn()
}
```

And make `getPatientOrFail` to return that mocked patient. But NABD.

I agree with removing that `makePatient() as PatientModel`, that was hacky. 😅

Reply:
`getPatientOrFail`returns a PatientModel so unless i do the as PatientModel then i need to also append all the other attributes to the mockPatient